karton/archive_extractor/archive_extractor.py

```diff
@@ -67,20 +70,28 @@ def log_message_wrapped(message: str, *args, **kwargs) -> None:
             self.log.warning("Failed to load as PE file.")
             return None
 
-        with tempfile.NamedTemporaryFile() as f:
+        # we need to use a temporary directory because debloat can implicitly unpack
+        # NSIS archives to parent directory of the passed file
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            temp_file = Path(tmp_dir) / filename
+
             process_pe(
                 pe,
-                out_path=f.name,
+                out_path=temp_file.as_posix(),
                 unsafe_processing=False,
```

Comment:
debloat 1.5.0 renamed `unsafe_processing` param to `last_ditch_processing` (https://github.com/Squiblydoo/debloat/commit/653693faa19471aea227c883539cbebc7592b7d4)
```suggestion
                last_ditch_processing=False,
```